File: src/board.cpp
#include "board.hpp"
#include <conio.h>
#include <windows.h>


Board::Board()
{
	w=defaultsize;
	h=defaultsize;
	data = new string*[h];
	for(int i=0;i<h;i++)
	{
		data[i]=new string[w];
	}
	for(int i=0;i<h;i++)
	{
		for(int j=0;j<w;j++)
		{
			data[i][j]="";
		}
	}

}

Board::Board(int _w, int _h)
{
	w=_w;
	h=_h;
	data= new string*[h];
	for(int i=0;i<h;i++)
	{
		data[i]=new string[w];
	}
	for(int i=0;i<h;i++)
	{
		for(int j=0;j<w;j++)
		{
			data[i][j]="";
		}
	}

}

Board::Board(const Board& B)
{
	w=B.w;
	h=B.h;
	data = new string*[h];
	for(int i=0;i<h;i++)
	{
		data[i]=new string[w];
	}

	for(int i=0;i<h;i++)
	{
		for(int j=0;j<w;j++)
		{
			data[i][j]=B.data[i][j];
		}
	}

}

Board::~Board()
{
	w=0;
	h=0;
	delete data;
}
Board& Board::operator=(const Board& B)
{
	delete data;
	w=B.w;
	h=B.h;
	data = new string*[h];
	for(int i=0;i<h;i++)
	{
		data[i]=new string[w];
	}
	for(int i=0;i<h;i++)
	{
		for(int j=0;j<w;j++)
		{
			data[i][j]=B.data[i][j];
		}
	}

	return *this;
}

const int Board::GetW()
{
	return w;
}

const int Board::GetH()
{
	return h;
}

string Board::GetEl(int _h,int _w)
{
	return data[_h][_w];
}

void Board::SetEl(char c,int _h,int _w)
{
	data[_h][_w]+=c;
}
void Board::DelEl(char c,int _h,int _w)
{
	string s=data[_h][_w];
	for(int i=0;i<s.length();i++)
	{
		if(s[i]==c)
		{
			data[_h][_w].erase(i,1);
			break;
		}
	}
}

void gotoxy( int column, int line )
  {
  COORD coord;
  coord.X = column;
  coord.Y = line;
  SetConsoleCursorPosition(
    GetStdHandle( STD_OUTPUT_HANDLE ),
    coord
    );
  }

void Board::PrintBoard()
{
	for(int i=0;i<h;i++)
	{
        gotoxy(0,i);
		for(int j=0;j<w;j++)
		{
		    if( data[j][i].length() > 0 )
                cout << data[j][i][0];
            else
                cout << " ";
		}
		cout << endl;
	}
}

bool Board::isEmpty() {
    for(int i=0;i<h;i++)
	{
		for(int j=0;j<w;j++)
		{
		    if( GetEl(j,i).length() > 0 )
                return false;
		}
	}
	return true;
}

File: jsrc/Board.java
class Board {
    private static final int defaultsize = 10;
    private String[][] data;
    private int w;
    private int h;

    public Board() {
        w = defaultsize;
        h = defaultsize;
        data = new String[h][w];
    }

    public Board(int _w, int _h) {
        w = _w;
        h = _h;
        data = new String[h][w];
    }

    public int GetW() {
        return w;
    }

    public int GetH() {
        return h;
    }

    public String GetEl(int _h, int _w) {
        return data[_h][_w];
    }

    public void SetEl(char c, int _h, int _w) {
        data[_h][_w] += c;
    }

    public void DelEl(char c, int _h, int _w) {
        StringBuilder sb = new StringBuilder(data[_h][_w]);
        for (int i = 0; i < sb.length(); i++) {
            if (sb.charAt(i) == c) {
                sb.deleteCharAt(i);
                break;
            }
        }
        data[_h][_w] = sb.toString();
    }

    public void PrintBoard() {
        for (int i = 0; i < h; i++) {
            for (int j = 0; j < w; j++) {
                if (data[j][i].length() > 0)
                    System.out.print(data[j][i].charAt(0));
                else
                    System.out.print(" ");
            }
            System.out.println();
        }
    }

    public boolean isEmpty() {
        for (int i = 0; i < h; i++) {
            for (int j = 0; j < w; j++) {
                if (data[i][j].length() > 0)
                    return false;
            }
        }
        return true;
    }
}